lib/pwa.js

```diff
@@ -251,37 +251,61 @@ exports.findByEncodedStartUrl = function(encodedStartUrl) {
 };
 
 /**
- * Validates and saves the PWA
+ * Process PWA
+ *
+ * Steps:
+ *  1) Validate PWA
+ *  2) Update Pwa's Manifest and save
+ *  3) Update PWA's MetadataDescription & Icon and save
+ *  4) (in background) Update Lighthouse & send notifications for new PWAs.
  *
  * @param {Pwa} pwa to update
  * @return {Pwa} the updated PWA
  */
-exports.save = function(pwa) {
-  if (!pwa) {
-    return Promise.reject('pwa is null');
+exports.process = function(pwa) {
+  const valid = this.validate(pwa);
+  if (!valid) {
+    return Promise.reject(valid);
   }
+  // remove hash from url
+  pwa.manifestUrl = pwa.manifestUrl.replace(/#.*/, '');
```

Comment:
Maybe I don't understand the use case. When does it make sense to have a hash in a manifest URL?